runtime/vm/createramclass.cpp

```diff
@@ -197,6 +197,22 @@ static void initializeClassLinks(J9Class *ramClass, J9Class *superclass, J9Memor
 #define MAGIC_ACCESSOR_IMPL "jdk/internal/reflect/MagicAccessorImpl"
 #endif /* JAVA_SPEC_VERSION == 8 */
 
+static J9Class *
+#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
+createdArrayClass(J9Class *elementClass, UDATA valueTypeFlags) {
```

Comment:
`getArrayClass` is a better name.

This can be turned into a macro that takes a J9Class and a UDATA valueTypeFlags. We can always pass in 0 as valueTypeFlags in non-VT build.

Reply:
@hangshao0 out of curiosity, what is your reasoning behind preferring a macro to something like the following?

```
static inline J9Class *
getArrayClass(J9Class *elementClass, UDATA valueTypeFlags = 0)
{
#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
        if (J9_ARE_ALL_BITS_SET(valueTypeFlags, J9ClassArrayIsNullRestricted)) {
                return elementClass->nullRestrictedArrayClass;
        }
#endif /* defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES) */
        return elementCalss->arrayClass;
}
```

Reply:
Macro is usually cheaper that a function call.

Reply:
True, but if it is specified as an `inline` function, given it's body, it would likely be inlined.